assets/js/base/context/providers/cart-checkout/checkout-processor.js

```diff
@@ -211,34 +211,51 @@ const CheckoutProcessor = () => {
 				}
 				return response.json();
 			} )
-			.then( ( response ) => {
-				dispatchActions.setAfterProcessing( response );
+			.then( ( responseJson ) => {
+				dispatchActions.setAfterProcessing( responseJson );
 				setIsProcessingOrder( false );
 			} )
-			.catch( ( fetchResponse ) => {
-				processCheckoutResponseHeaders(
-					fetchResponse.headers,
-					dispatchActions
+			.catch( ( errorResponse ) => {
+				let errorNotice = sprintf(
+					// Translators: %s Error text.
+					__(
+						'Error: %s Please try placing your order again.',
+						'woo-gutenberg-products-block'
+					),
+					errorResponse?.message ??
+						__(
+							'Something went wrong.',
+							'woo-gutenberg-products-block'
+						)
 				);
-				fetchResponse.json().then( ( response ) => {
-					// If updated cart state was returned, update the store.
-					if ( response.data?.cart ) {
-						receiveCart( response.data.cart );
+
+				try {
+					if ( errorResponse?.headers ) {
+						processCheckoutResponseHeaders(
+							errorResponse.headers,
+							dispatchActions
+						);
 					}
-					addErrorNotice( formatStoreApiErrorMessage( response ), {
-						id: 'checkout',
-					} );
-					response.additional_errors?.forEach?.(
-						( additionalError ) => {
-							addErrorNotice( additionalError.message, {
-								id: additionalError.error_code,
-							} );
+					// This attempts to parse a JSON error response where the status code was 4xx/5xx.
+					errorResponse.json().then( ( response ) => {
+						// If updated cart state was returned, update the store.
+						if ( response.data?.cart ) {
+							receiveCart( response.data.cart );
 						}
-					);
-					dispatchActions.setHasError( true );
-					dispatchActions.setAfterProcessing( response );
-					setIsProcessingOrder( false );
-				} );
+						errorNotice = formatStoreApiErrorMessage( response );
```

Comment:
I don't understand the reason behind declaring errorNotice and replacing it here. What changes?

Reply:
Just to avoid calling addErrorNotice in multiple places. it's done at the end.